internal/juju/secrets.go

```diff
@@ -238,3 +258,57 @@ func (c *secretsClient) DeleteSecret(input *DeleteSecretInput) error {
 
 	return nil
 }
+
+// GrantSecretAccess grants access to a secret.
+func (c *secretsClient) GrantSecretAccess(input *GrantRevokeSecretAccessInput) error {
+	conn, err := c.GetConnection(&input.ModelName)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = conn.Close() }()
+
+	secretAPIClient := c.getSecretAPIClient(conn)
+
+	secretURI, err := coresecrets.ParseURI(input.SecretId)
+	if err != nil {
+		return err
+	}
+
+	// In the current implementation, the secret name is not used.
+	results, err := secretAPIClient.GrantSecret(secretURI, "", input.Applications)
+	if err != nil {
+		return typedError(err)
+	}
+	if results[0] != nil && len(results) > 1 {
+		return &MultiError{Errors: results}
+	}
+
+	return nil
+}
+
+// RevokeSecretAccess revokes access to a secret.
+func (c *secretsClient) RevokeSecretAccess(input *GrantRevokeSecretAccessInput) error {
```

Comment:
These functions are 99% identical, so they should be abstracted away. The api functions can be passed as a parameter,  see: https://github.com/juju/juju/blob/a7538882134dd2cc0398bb5dac41655e5f4265d4/apiserver/facades/client/secrets/secrets.go#L531